assets/quick-order-list.js

```diff
@@ -1,307 +1,279 @@
+          button.addEventListener('click', (event) => {
+            event.preventDefault();
+            this.toggleLoading(true);
+            this.startQueue(button.dataset.index, 0);
+          });
+        });
+      }
+
+      get currentPage() {
+        return this.querySelector('.pagination-wrapper')?.dataset?.page ?? '1';
+      }
+
+      get cartVariantsForProduct() {
+        return JSON.parse(this.querySelector('[data-cart-contents]')?.innerHTML || '[]');
       }
 
       onChange(event) {
         const inputValue = parseInt(event.target.value);
         this.cleanErrorMessageOnType(event);
         if (inputValue == 0) {
+          event.target.setAttribute('value', inputValue);
           this.startQueue(event.target.dataset.index, inputValue);
         } else {
           this.validateQuantity(event);
         }
       }
 
       cleanErrorMessageOnType(event) {
-        event.target.addEventListener('keydown', () => {
+        const handleKeydown = () => {
           event.target.setCustomValidity(' ');
           event.target.reportValidity();
-        });
+          event.target.removeEventListener('keydown', handleKeydown);
+        };
+
+        event.target.addEventListener('keydown', handleKeydown);
       }
 
       validateInput(target) {
+        const targetValue = parseInt(target.value);
+        const targetMin = parseInt(target.dataset.min);
+        const targetStep = parseInt(target.step);
+
         if (target.max) {
           return (
-            parseInt(target.value) == 0 ||
-            (parseInt(target.value) >= parseInt(target.dataset.min) &&
-              parseInt(target.value) <= parseInt(target.max) &&
-              parseInt(target.value) % parseInt(target.step) == 0)
+            targetValue == 0 ||
+            (targetValue >= targetMin && targetValue <= parseInt(target.max) && targetValue % targetStep == 0)
           );
         } else {
-          return (
-            parseInt(target.value) == 0 ||
-            (parseInt(target.value) >= parseInt(target.dataset.min) &&
-              parseInt(target.value) % parseInt(target.step) == 0)
-          );
+          return targetValue == 0 || (targetValue >= targetMin && targetValue % targetStep == 0);
         }
       }
 
-      refresh() {
-        return new Promise((resolve, reject) => {
-          fetch(`${this.getSectionsUrl()}?section_id=${this.sectionId}`)
-            .then((response) => response.text())
-            .then((responseText) => {
-              const html = new DOMParser().parseFromString(responseText, 'text/html');
-              const sourceQty = html.querySelector(`#${this.quickOrderListId}`);
-              if (sourceQty) {
-                this.innerHTML = sourceQty.innerHTML;
-              }
-              resolve();
-            })
-            .catch((e) => {
-              console.error(e);
-              reject(e);
-            });
-        });
-      }
-
       getSectionsToRender() {
```

Comment:
Sections / targets got updated to [simplify cart update the same as in quick-add-bulk](https://github.com/Shopify/dawn/pull/3710/files#r1937325971) and remove the duplicate section for quick order list.